ir/src/main/scala/org/scalajs/core/ir/Printers.scala

```diff
@@ -1052,4 +1052,601 @@ object Printers {
+
+        case VarDef(ident, vtpe, mutable, rhs) =>
+          println("VarDef (")
+          indent()
+          println(ident)
+          println(vtpe)
+          println("mutable = " + mutable.toString)
+          print(rhs); paren()
+          undent()
+
+
+        case ParamDef(ident, ptpe, mutable, rest) =>
+          println("ParamDef (")
+          indent()
+          println(ident)
+          println(ptpe)
+          println("mutable = " + mutable.toString)
+          print("rest = " + rest.toString); paren()
+          undent()
+
+        // Control flow constructs
+
+        case Skip() =>
+          print("Skip")
+
+        case Block(trees) =>
+          printList(trees)
+
+        case Labeled(label, tpe, body) =>
+          println("Labeled (")
+          indent()
+          println(tpe)
+          print(body); paren()
+          undent()
+
+        case Assign(lhs, rhs) =>
+          println("Assign (")
+          indent()
+          println(lhs)
+          print(rhs); paren()
+          undent()
+
+        case Return(expr, label) =>
+          println("Return (")
+          indent()
+          println(expr)
+          label match {
+            case Some(l) => print(l)
+            case None    => print("None")
+          }
+          paren()
+          undent()
+
+        case If(cond, thenp, elsep) =>
+          println("If (")
+          indent()
+          println(cond)
+          println(thenp)
+          print(elsep); paren()
+          undent()
+
+        case While(cond, body, label) =>
+          println("While (")
+          indent()
+          println(cond)
+          print(body); paren()
+          undent()
+
+        case DoWhile(body, cond, label) =>
+          println("DoWhile (")
+          indent()
+          println(body)
+          println(cond)
+          label match {
+            case Some(l) => print(l)
+            case None    => print("None")
+          }
+          paren()
+          undent()
+
+        case Try(block, errVar, handler, finalizer) =>
+          println("Try (")
+          indent()
+          println(block)
+          println(errVar)
+          println(handler)
+          print(finalizer); paren()
+          undent()
+
+        case Throw(expr) =>
+          println("Throw (")
+          indent()
+          print(expr); paren()
+          undent()
+
+        case Continue(label) =>
+          println("Continue (")
+          indent()
+          label match {
+            case Some(l) => print(l)
+            case None    => print("None")
+          }
+          paren()
+          undent()
+
+        case Match(selector, cases, default) =>
+          println("Match (")
+          indent()
+          println(selector)
+
+          cases foreach { pair =>
+            print(pair._1.toString()); println(":")
+            indent(); print(pair._2); undent()
+          }
+
+          print(default); paren()
+          undent()
+
+
+        case Debugger() =>
+          print("Debugger")
```

Comment:
Explicitly write the `()`. Otherwise it looks like a `case object` instead of a `case class`